### ISSUE TITLE
[Tablet orders] Deeplinked order that is not originally in the order list isn't selected in the order list

### ISSUE THREAD

**Issue by jaclync** [woocommerce/woocommerce-ios#11808]
**Describe the bug**

In multi-column split view - when an order is deep linked from either an order link or push notification, if this order doesn't exist locally (like it hasn't been fetched before from pagination or a new order) then the order cell isn't highlighted in the order list.

**To Reproduce**

Prerequisite: the account is connected to more than one store

- Launch the app in the multi-column state on a wide device
- Prepare an order link in a separate app
- Switch to a different store from the store ID in the link
- Tap on an order universal link --> the order in the link is shown, but the row isn't highlighted in the order list

**Screenshots**

https://github.com/woocommerce/woocommerce-ios/assets/1945542/0ed7a942-4aec-438b-bd21-5b6a06cacc5f

**Expected behavior**

The order row should be highlighted in the order list pane.

**Isolating the problem (mark completed items with an [x]):**
- [ ] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [ ] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [x] I can reproduce this bug consistently using the steps above.

**Mobile Environment**
Please include:
- Device: iPad Air 5th gen simulator / iPad Pro 11' device
- iOS version: iOS 17.2
- WooCommerce iOS version: 17.0

**Comment by joshheald** [comment]
Automated triage note: this was tested and commented by Codex using device/simulator automation. Could not validly retest on iPad Air (M1) running iPadOS 18.7.1, Woo (Dev) 24.9 (24.9.0.4) at df01963e. I confirmed the Orders split-view state, but the report requires a prepared universal link to an order not already present locally plus switching to a different store before opening the link. This lane did not have the target order link or safe multi-store setup, so the deeplink selection behavior needs a prepared handoff.

**Comment by dangermattic** [comment]
Thanks for reporting! 👍

**Comment by jaclync** [comment]
Closing due to low priority - I tested and reproduced the issue on a tablet device in TestFlight 25.1 build, but order details were loaded as expected at the correct store and the highlighted state in the order list pane felt like a nice-to-have.
